service/matching/matcher.go

```diff
@@ -249,6 +262,13 @@ func (tm *TaskMatcher) OfferQuery(ctx context.Context, task *internalTask) (*mat
 				continue
 			}
 			return nil, err
+		case <-noPollerCtxC:
+			// only error if there has not been a recent poller. Otherwise, let it wait for the remaining time
+			// hopping for a match, or ultimately returning the default CDE error.
+			if tm.timeSinceLastPoll() > tm.config.QueryPollerUnavailableWindow() {
```

Comment:
Should we only do this on the root partition?
It's more likely to see pollers due to forwarding.